ITISFoundation/osparc-simcore
# ✨ Error pages when published studies dispatcher fails, access rights fixes and plugin refactoring

## What do these changes do?

To publish a study it needs to:
1. Be a template
2. Share with everybody
3. ``published`` fiend in the database must be checked

Currently any published study can be accessed via a link with path ``/study/{project_id}``. That link will copy and open the study even if the user is not logged it.  If this request fails it will now display an error page with details.

This PR provides:
- ✨ Improves error handling and reporting. Redirects to an error page created in the front-end (previously was just raw text)
- ✨ Guest users (i.e. the ones that are not registered) use now temp accounts with expiration date
- 🐛 Fixes permission errors while copying data from published template

@odeimaiz some improvements in the FE are necessary for the two first items -> #3987

In addition, this PR
- ♻️ refactors error handling and tests. (follow up in https://github.com/ITISFoundation/osparc-simcore/issues/3977 )
- 🔨 new action ``"Python: Remote Attach GC webserver"`` in the `launch.template.json`` to debug the garbage collection service
- 🐛 small fix in catalog

## Related issue/s

- Related to https://github.com/ITISFoundation/osparc-issues/issues/829: error handling
- Related to https://github.com/ITISFoundation/osparc-simcore/issues/3951
- Fixes https://github.com/ITISFoundation/osparc-simcore/issues/3884: fixes permission errors while copying data


## How to test

1. Publish a study (follow the definition above to publish a study). browse ``https://{host}:{port}/studies/{study_uuid}`` as guest and will open a copy of the study
2. log in. browse ``https://{host}:{port}/studies/{study_uuid}`` and will open a copy of the study
3. disable publish flag. browse ``https://{host}:{port}/studies/{study_uuid}`` should fail with an error page
4. Enable publish flag and remove everyone access rights.  browse ``https://{host}:{port}/studies/{study_uuid}`` should fail with an error page
5.  Enable publish flag and remove everyone access rights, then browse ``https://{host}:{port}/studies/{study_uuid}`` should fail with an error page

![Pasted image 20230316142241](https://user-images.githubusercontent.com/32402063/226030479-06c75d29-5826-4c08-8eb0-49d8accc6c90.png)


```
cd services/web/server
make install-dev
pytest -vv tests/unit/with_dbs/01/test_studies_dispatcher_studies_access.py
```

## Checklist

- [x] ``make version-*``
